x/escrow/keeper/internal/genesis.go

```diff
@@ -0,0 +1,368 @@
+package internal
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	escrowv1alpha1 "github.com/0tech/andromeda/x/escrow/andromeda/escrow/v1alpha1"
+)
+
+func (k Keeper) DefaultGenesis() *escrowv1alpha1.GenesisState {
+	return &escrowv1alpha1.GenesisState{
+		Params:       DefaultGenesisParams(),
+		NextAgent:    1,
+		Agents:       []*escrowv1alpha1.GenesisState_Agent{},
+		NextProposal: 1,
+		Proposals:    []*escrowv1alpha1.GenesisState_Proposal{},
+	}
+}
+
+func DefaultGenesisParams() *escrowv1alpha1.GenesisState_Params {
+	return &escrowv1alpha1.GenesisState_Params{
+		MaxMetadataLength: 255,
+	}
+}
+
+func (k Keeper) ValidateGenesis(gs *escrowv1alpha1.GenesisState) error {
+	if gs.Params == nil {
+		return escrowv1alpha1.ErrUnimplemented.Wrap("nil params")
+	}
+
+	if gs.NextAgent == 0 {
+		return escrowv1alpha1.ErrUnimplemented.Wrap("nil next_agent")
+	}
+
+	if gs.Agents == nil {
+		return escrowv1alpha1.ErrUnimplemented.Wrap("nil agents")
+	}
+
+	if gs.NextProposal == 0 {
+		return escrowv1alpha1.ErrUnimplemented.Wrap("nil next_proposal")
+	}
+
+	if gs.Proposals == nil {
+		return escrowv1alpha1.ErrUnimplemented.Wrap("nil proposals")
+	}
+
+	if err := k.validateGenesisParams(gs.Params); err != nil {
+		return errorsmod.Wrap(err, "params")
+	}
+
+	if err := k.validateGenesisAgents(gs.Agents); err != nil {
+		return errorsmod.Wrap(err, "agents")
+	}
+
+	if err := k.validateGenesisProposals(gs.Proposals); err != nil {
+		return errorsmod.Wrap(err, "proposals")
+	}
+
+	return nil
```

Comment:
Actually, the former is more specific error, because `ErrInvalidGenesis` means there is something invalid in genesis, while `ErrUnimplemented` specifies the reason is providing nil params, and it is unimplemented now.